packages/core-ethereum/src/indexer/index.ts

```diff
@@ -400,23 +411,44 @@ class Indexer extends EventEmitter {
           this.chain.updateConfirmedTransaction(nativeTx)
         })
       }
-
-      if (fetchEvents) {
-        const events = await this.getEvents(
-          blockNumber - this.maxConfirmations,
-          blockNumber - this.maxConfirmations,
-          true
-        )
-
-        this.onNewEvents(events)
-      }
     } catch (err) {
       log(
         `error: failed to retrieve information about block ${blockNumber} with finality ${this.maxConfirmations}`,
         err
       )
     }
 
+    if (fetchEvents) {
+      const RETRIES = 3
+      let events: TypedEvent<any, any>[]
+
+      for (let i = 0; i < RETRIES; i++) {
+        try {
+          events = await this.getEvents(blockNumber - this.maxConfirmations, blockNumber - this.maxConfirmations, true)
+        } catch (err) {
+          if (i < RETRIES) {
+            // Give other tasks CPU time to happen
+            // Push next provider query to end of next event loop iteration
+            await setImmediate()
+            continue
+          } else {
+            log(
+              `Cannot fetch block ${blockNumber - this.maxConfirmations} despite ${RETRIES} retries. Skipping block.`,
+              err
+            )
```

Comment:
We should throw again after logging, otherwise the semantics of this is broken.

Reply:
In this case we need to catch the error somewhere else - which is not the case. Therefore the error is logged because we can live without a block but we don't want to terminate because of unhandled promise rejections on every missing blccked. This is why the error is logged after > 1 retries and not thrown